counterpartylib/test/arc4_test.py

```diff
@@ -0,0 +1,142 @@
+import pprint
+import tempfile
+import pytest
+from counterpartylib.test import conftest  # this is require near the top to do setup of the test suite
+from counterpartylib.test import util_test
+from counterpartylib.test.util_test import CURR_DIR
+from counterpartylib.test.fixtures.params import DP, ADDR
+
+from counterpartylib.lib import util
+from counterpartylib.lib import arc4
+from counterpartylib.lib.messages import send
+from counterpartylib.lib import transaction
+
+FIXTURE_SQL_FILE = CURR_DIR + '/fixtures/scenarios/unittest_fixture.sql'
+FIXTURE_DB = tempfile.gettempdir() + '/fixtures.unittest_fixture.db'
+
+
+def test_arc4_mocked():
+    """
+    by default init_arc4 should be mocked in the test suite to always use `'00' * 32` as seed.
+     so when we use `'01' * 32` as seed it should still return the same result as `'00' * 32` or as ``
+    """
+    text = bytes("testing", 'utf-8')
+    
+    # '00' * 32 encrypt
+    k = arc4.init_arc4('00' * 32)
+    assert k.encrypt(text) == b'\xaa}\xfa5\xcaY:'
+
+    # '00' * 32 decrypt
+    k = arc4.init_arc4('00' * 32)
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+
+    # b'\x00' * 32 encrypt
+    k = arc4.init_arc4(b'\x00' * 32)
+    assert k.encrypt(text) == b'\xaa}\xfa5\xcaY:'
+    
+    # b'\x00' * 32 decrypt
+    k = arc4.init_arc4(b'\x00' * 32)
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+    
+    # '' * 32 encrypt
+    k = arc4.init_arc4('' * 32)
+    assert k.encrypt(text) == b'\xaa}\xfa5\xcaY:'
+
+    # '' * 32 decrypt
+    k = arc4.init_arc4('' * 32)
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+
+    # b'' * 32 encrypt
+    k = arc4.init_arc4(b'' * 32)
+    assert k.encrypt(text) == b'\xaa}\xfa5\xcaY:'
+    
+    # b'' * 32 decrypt
+    k = arc4.init_arc4(b'' * 32)
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+    
+    # '01' * 32 encrypt
+    k = arc4.init_arc4('01' * 32)
+    assert k.encrypt(text) == b'\xaa}\xfa5\xcaY:'
+
+    # '01' * 32 decrypt
+    k = arc4.init_arc4('01' * 32)
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+
+    # b'\x01' * 32 encrypt
+    k = arc4.init_arc4(b'\x01' * 32)
+    assert k.encrypt(text) == b'\xaa}\xfa5\xcaY:'
+    
+    # b'\x01' * 32 decrypt
+    k = arc4.init_arc4(b'\x01' * 32)
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+
+
+@conftest.add_fn_property(DISABLE_ARC4_MOCKING=True)
```

Comment:
still a way to disable the mocking in the test suite and still test the actual results.